mailbox/backup/src/main/java/org/apache/james/mailbox/backup/MailBoxAnnotationsExtraField.java

```diff
@@ -0,0 +1,56 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.backup;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.commons.compress.archivers.zip.ZipShort;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+
+import com.google.common.collect.ImmutableList;
+
+public class MailBoxAnnotationsExtraField extends StringExtraField {
+
+    public static final ZipShort ID_AQ = new ZipShort(0x7161); // "aq" in little-endian
+
+    private static String serializeAnnotations(List<MailboxAnnotation> annotations) {
+        return annotations.stream().map(MailBoxAnnotationsExtraField::serializeAnnotation)
+            .collect(Collectors.joining("%"));
```

Comment:
what is `%`?

Reply:
It's the separator between different annotations. I used the same as used in the flags serialization.
If we agree to use the same, we should make it a constant.

Reply:
can you check the RFC if there is any character that we can safely use for that?

Reply:
@rouazana do you have a link to the RFC in question ?

Reply:
should be https://datatracker.ietf.org/doc/rfc5464/?include_text=1

Reply:
The '%' character is safe to use :
Entry names MUST NOT contain asterisk ("*") or percent ("%")
   characters and MUST NOT contain non-ASCII characters or characters
   with octet values in the range 0x00 to 0x19.  Invalid entry names
   result in a BAD response in any IMAP command in which they are used.

Reply:
entry seems to be about the key, what about the value?

Reply:
(if there is no limitation, maybe we should encode value, for example with base64?)

Reply:
base64 encoding will further decrease the available size for annotations.
What data will be stored in annotations ? 
What would be preferable to encode the the data and truncate the data earlier or to choose to replace in the extra field the character choosed to spit the annotations by another character ( a space ? ) ?

Reply:
the issue is that I don't see any safe character to split annotations. Even \0 seems to be allowed. I'm not sure annotation size is so much an issue, but we could use a more economic format than base64, such as base122 for example.

Reply:
OR we can store the annotations has file entry in the mailbox directory, the name of each annotation could be the entry name.
This way we will not encounter data truncation issue, has a bonus the content will be compressed.